### PR TITLE
install contemporary git and clang

### LINKED ISSUE #5: update some tools in the builder image
> - wget
> - clang (more recent release than what comes by default)
> - more recent gcc
> - more recent ninja -- from https://github.com/ninja-build/ninja/releases
> 

from: https://github.com/openziti/ziti-cmake/pull/1#pullrequestreview-1439579921

### PR DESCRIPTION
**Pull request by qrkourier**
resolves #5

### CODE REVIEW

**Comment by ekoby** [comment]
couple more tools:
- clang-tidy (from llvm repo)
- cppcheck
- gcov

**Comment by qrkourier** [comment]
> * gcov

@ekoby `gcovr`?

File: docker-image/Dockerfile [hunk 1]
```diff
@@ -4,51 +4,82 @@
...
+    && apt-get clean autoclean \
+    && rm -fr /var/lib/apt/lists/{apt,dpkg,cache,log} /tmp/* /var/tmp/*
+
+# provides patch releases for 1.11 in /usr/local/bin/ninja
+RUN python3 -m pip install ninja~=1.11.0
+
+RUN curl -sSLf https://apt.llvm.org/llvm-snapshot.gpg.key \
+    | gpg --dearmor --output /usr/share/keyrings/llvm-snapshot.gpg \
+    && chmod +r /usr/share/keyrings/llvm-snapshot.gpg \
+    && echo "deb [signed-by=/usr/share/keyrings/llvm-snapshot.gpg] http://apt.llvm.org/bionic/ llvm-toolchain-bionic-7 main" > /etc/apt/sources.list.d/llvm-snapshot.list
```
[the remark this answers is not in the document]
Comment (qrkourier) [reply docker-image/Dockerfile:71]: No good reason to use v7, and I confirmed v17 is available from the Bionic repo.

**Comment by qrkourier** [comment]
Relevant discussion about sticking with gcc v7 on Bionic vs compiling gcc v11 or v13 from source in this thread: https://github.com/openziti/ziti-builder/issues/5#issuecomment-1688142202

**Comment by qrkourier** [comment]
I pushed this branch's local container image build to `docker.io/kbinghamnetfoundry/ziti-builder:latest` and tested the ZET's main cmake.yml workflow here: https://github.com/openziti/ziti-tunnel-sdk-c/actions/runs/5931869807?pr=716

File: docker-image/Dockerfile [hunk 2]
```diff
@@ -80,6 +113,21 @@ RUN cd /usr/local \
     && ./vcpkg/bootstrap-vcpkg.sh -disableMetrics \
     && chmod -R ugo+rwX /usr/local/vcpkg
 
+# RUN cd /usr/local/src \
```
Comment (scareything) [inline docker-image/Dockerfile:116]: We could run these build steps (and any apt installs that are only needed for gcc) in a [stage](https://docs.docker.com/build/building/multi-stage/) and then `COPY` the install directory into the final stage of this image. This makes cleanup perfect and effortless.

Comment (qrkourier) [reply docker-image/Dockerfile:116]: I agree that will eliminate any compilation-related bloat in the final image, if we decide to go the route of compiling the three gcc v11 or v13 suites for native, cross-arm, cross-arm64.

**Review by ekoby — APPROVED** [review]

**Pull request merged**